pkg/systemd/parser/unitfile.go

```diff
@@ -376,53 +375,39 @@ func (p *UnitFileParser) flushPendingComments(toComment bool) {
 	}
 }
 
-func nextLine(data string, afterPos int) (string, string) {
-	rest := data[afterPos:]
-	if i := strings.Index(rest, "\n"); i >= 0 {
-		return strings.TrimSpace(data[:i+afterPos]), data[i+afterPos+1:]
-	}
-	return data, ""
-}
-
-func trimSpacesFromLines(data string) string {
-	lines := strings.Split(data, "\n")
-	for i, line := range lines {
-		lines[i] = strings.TrimSpace(line)
-	}
-	return strings.Join(lines, "\n")
-}
-
 // Parse an already loaded unit file (in the form of a string)
 func (f *UnitFile) Parse(data string) error {
 	p := &UnitFileParser{
-		file:   f,
-		lineNr: 1,
+		file: f,
 	}
 
-	data = trimSpacesFromLines(data)
-
-	for len(data) > 0 {
-		origdata := data
-		nLines := 1
-		var line string
-		line, data = nextLine(data, 0)
+	lines := strings.Split(strings.TrimSuffix(data, "\n"), "\n")
+	remaining := ""
 
-		if !lineIsComment(line) {
-			// Handle multi-line continuations
-			// Note: This doesn't support comments in the middle of the continuation, which systemd does
-			if lineIsKeyValuePair(line) {
-				for len(data) > 0 && line[len(line)-1] == '\\' {
-					line, data = nextLine(origdata, len(line)+1)
-					nLines++
+	for lineNr, line := range lines {
+		if lineIsComment(line) {
+			// ignore the comment is inside a continuation line.
+			if remaining != "" {
+				continue
+			}
+		} else {
+			line = strings.TrimSpace(line)
+			if strings.HasSuffix(line, "\\") {
+				line = line[:len(line)-1]
+				// check whether the line is a continuation of the previous line
```

Comment:
Sorry for not being clear. Doesn't this line belong above the `if remaining != "" {` line?

Reply:
yeah that is better.  I originally added it to `if strings.HasSuffix(line, "\\")` but using it where you suggest is better.